sql/query/insert_test.go

```diff
@@ -128,4 +128,53 @@ func TestInsertStmt(t *testing.T) {
 		require.NoError(t, err)
 		require.JSONEq(t, `{"a": "a", "b-b": "b"}`, buf.String())
 	})
+
+	t.Run("with types constraints", func(t *testing.T) {
```

Comment:
This is great! There is only one thing missing: making sure when we insert something that doesn't follow the constraint, we have an error. For constraints with and without `NOT NULL`.

Reply:
Good call! I've added a test for this in 27a1b2a. I'm wondering if it's enough or should I add a test for each constraint type. What do you think?

Reply:
I think you can easily create a table-driven test for each type, each tech would only need to replace the type in the query. ex: `db.Exec("CREATE TABLE foo (a "+ test.FieldConstraint +")")`

Reply:
I pushed more test in 300447e.